addons/mass_mailing/wizard/mailing_list_add_contact.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class MailingListAddContact(models.TransientModel):
+    _name = "mailing.list.add.contact"
+    _description = "Add contact to mailing list wizard"
+
+    contact_ids = fields.Many2many('mailing.contact')
+    mailing_list_id = fields.Many2one('mailing.list', string='Mailing List', required=True)
+
+    def action_add_contact_to_mailing_list(self):
+        self.ensure_one()
+        ctx = dict(self.env.context)
+        ctx.pop('default_state', None)
+        self = self.with_context(ctx)
+
+        for contact_id in self.contact_ids:
+            self.mailing_list_id.write({
+                'contact_ids': [(4, contact_id.id, 0)]
+            })
```

Comment:
As a general rule, we must really avoid database operations (create/write/unlink) in loops.
You can easily add a list of commands to execute in the o2m like so:

```
self.mailing_list_id.write({
    'contact_ids': [(4, contact.id) for contact in self.contact_ids]
})
```